core/src/main/java/org/elasticsearch/monitor/fs/FsInfo.java

```diff
@@ -188,12 +189,302 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
+            this.currentReadMilliseconds = currentReadMilliseconds;
+            this.previousReadMilliseconds = previousReadMilliseconds;
+            this.currentWriteMilliseconds = currentWriteMilliseconds;
+            this.previousWriteMilliseconds = previousWriteMilliseconds;
+            this.currentWeightedMilliseconds = currentWeightedMilliseconds;
+            this.previousWeightedMilliseconds = previousWeightedMilliseconds;
+            this.currentRelativeTime = currentRelativeTime;
+            this.previousRelativeTime = previousRelativeTime;
+        }
+
+        public DeviceStats(StreamInput in) throws IOException {
+            majorDeviceNumber = in.readVInt();
+            minorDeviceNumber = in.readVInt();
+            deviceName = in.readString();
+            currentReadsCompleted = in.readLong();
+            previousReadsCompleted = in.readLong();
+            currentWritesCompleted = in.readLong();
+            previousWritesCompleted = in.readLong();
+            currentSectorsRead = in.readLong();
+            previousSectorsRead = in.readLong();
+            currentSectorsWritten = in.readLong();
+            previousSectorsWritten = in.readLong();
+            currentReadMilliseconds = in.readLong();
+            previousReadMilliseconds = in.readLong();
+            currentWriteMilliseconds = in.readLong();
+            previousWriteMilliseconds = in.readLong();
+            currentWeightedMilliseconds = in.readLong();
+            previousWeightedMilliseconds = in.readLong();
+            currentRelativeTime = in.readLong();
+            previousRelativeTime = in.readLong();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVInt(majorDeviceNumber);
+            out.writeVInt(minorDeviceNumber);
+            out.writeString(deviceName);
+            out.writeLong(currentReadsCompleted);
+            out.writeLong(previousReadsCompleted);
+            out.writeLong(currentWritesCompleted);
+            out.writeLong(previousWritesCompleted);
+            out.writeLong(currentSectorsRead);
+            out.writeLong(previousSectorsRead);
+            out.writeLong(currentSectorsWritten);
+            out.writeLong(previousSectorsWritten);
+            out.writeLong(currentReadMilliseconds);
+            out.writeLong(previousReadMilliseconds);
+            out.writeLong(currentWriteMilliseconds);
+            out.writeLong(previousWriteMilliseconds);
+            out.writeLong(currentWeightedMilliseconds);
+            out.writeLong(previousWeightedMilliseconds);
+            out.writeLong(currentRelativeTime);
+            out.writeLong(previousRelativeTime);
+        }
+
+        private long totalOperationsCompleted() {
+            return (currentReadsCompleted - previousReadsCompleted) + (currentWritesCompleted - previousWritesCompleted);
+        }
+
+        public float iops() {
```

Comment:
Do we really want `float` here for everything? Sorry if this is how other stats apis work, i really have no idea.

But I do think in this day, `double` should be the default unless there is a good reason? If i do a quick google, i see people talking about millions of iops, whether that is real or fantasy, that range is already getting pretty crappy for `float`